src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.Helpers.cs

```diff
@@ -38,12 +38,20 @@ internal static MemberAccessor MemberAccessor
         private static JsonTypeInfo CreateTypeInfoCore(Type type, JsonConverter converter, JsonSerializerOptions options)
         {
             JsonTypeInfo typeInfo = JsonTypeInfo.CreateJsonTypeInfo(type, converter, options);
-            typeInfo.NumberHandling = GetNumberHandlingForType(typeInfo.Type);
-            typeInfo.PreferredPropertyObjectCreationHandling = GetObjectCreationHandlingForType(typeInfo.Type);
 
-            if (typeInfo.Kind == JsonTypeInfoKind.Object)
+            if (GetNumberHandlingForType(typeInfo.Type) is { } numberHandling)
```

Comment:
Not directly related to the change at hand, but this addresses an inconsistency in how attribute configuration is mapped to contract metadata.

Reply:
IMO this was more readable before, why do we avoid calling the setter?

Reply:
We don't want to call the setter if not attribute is specified. This is consistent with what we do in the source generator.